wordpress-mobile/WordPress-iOS
# Persisting type for Post entities.

Currently the only differentiation in posts pulled into the app are whether or not they are of type `page` or `post`. However, moving forward with any support for querying API posts that may have a [custom post type](http://jetpack.me/support/custom-content-types/) will result in simply persisting posts as a `Post` entity, without any set value for what type of post it actually is.

In this PR we are now persisting the `type` of a post with the new attribute `postType`. `AbstractPosts` synced via `PostService` will have their `postType` set upon sync. The `PostListViewController` also accounts for `postType` within its fetch request predicate.

Note: On one hand persisting `postType` for `AbstractPost` is helpful, but also seems counterintuitive to our current type assumptions of which entity an object is, `Post` or `Page`. Perhaps we should instead create a child `Post` entity for custom post types as a `CustomPost` entity? Or perhaps we shouldn't have separate entites at all for `Post` and `Page`, and instead reflect all querying/predicates based on the new `postType` for `page`, `post`, or custom types.

Testing for this implementation:

1. Open a Site, view blog posts.
2. Blog posts should display, page and search as normal and among the different filters (no change).
3. Open pages for the site, pages should display as normally.

@aerych please review, no rush, and any thoughts on this?
